skbio/tree/tests/test_tree.py

```diff
@@ -1303,6 +1303,42 @@ def test_shuffle_raises(self):
         with self.assertRaises(MissingNodeError):
             next(self.simple_t.shuffle(names=['x', 'y']))
 
+    def test_cache_leafcount(self):
+        t = TreeNode.read([r'((a,b)e,(c,d)f,g)r;'])
+        t._cache_leafcount()
+        self.assertEquals(t.find('r').leafcount, 5)
```

Comment:
Note that these are helper methods -- I was planning on calling them whenever the coordinates are calculated (for rendering).  

If I'm understanding you - whenever `leafcount` is called, then it should call these helpers?  I'm not sure if I follow, especially for traversal up the the parent nodes.